modules/tinymce/src/core/main/ts/fmt/FormatUtils.ts

```diff
@@ -9,6 +9,49 @@ import * as NodeType from '../dom/NodeType';
 import * as Whitespace from '../text/Whitespace';
 import { BlockFormat, Format, FormatAttrOrStyleValue, FormatVars, InlineFormat, MixedFormat, SelectorFormat } from './FormatTypes';
 
+const internalAttributesPrefixes = [
+  'data-ephox-',
+  'data-mce-',
+  'data-alloy-',
+  'data-snooker-',
+  '_',
+  'aria-'
+];
+
+const internalDataValues = [
```

Comment:
Not all of these values are likely to be needed, for example `data-bedrockid` is probably not something we need to worry about ever showing up in production. But I thought it'd be better to add all of them at the start and then remove them when others point out they're probably not needed than to miss some. These values include the ones I could find from premium plugins. Many are likely attributes used very locally ( such as in snooker ) which never leaves it.

Reply:
I think I'd prefer to change them all to `data-mce` or similar. Some of these probably aren't used very often, and it's a lot of strings to bundle and add to the lookup function. Every one of these we can (and in my opinion should) update.

It may be a little annoying, but the long term effects seem worth it. Less code to maintain, and less ability for new data attributes to sneak in that aren't prefixed correctly.

Reply:
I agree with this, however, this would require us to also change values in the premium plugins ( some of these values are from them, and they use some of the others ) and they would not be fully backwards compatible against earlier versions, which I have been informed is not really acceptable. This is meant to allow us to be backwards compatible until we can fully fix this, presumably around 7.0...
Or at least this have been my understanding of the situation.

Reply:
That's... debatable. If we do what I suggested then new plugins will still work with old tinymce, but old plugins will start leaving data attributes around when used with new tinymce.

I'm not sure how much we really try to make compatibility work in that direction. Compatibility is more about making sure customer integrations continue to work after upgrading tinymce, not so much old versions of our own plugins.